certbot/certbot/_internal/client.py

```diff
@@ -438,7 +438,14 @@ def obtain_certificate(self, domains: List[str], old_keypath: Optional[str] = No
             csr = crypto_util.generate_csr(key, domains, self.config.csr_dir,
                                            self.config.must_staple, self.config.strict_permissions)
 
-        orderr = self._get_order_and_authorizations(csr.data, self.config.allow_subset_of_names)
+        try:
+            orderr = self._get_order_and_authorizations(csr.data, self.config.allow_subset_of_names)
+        except messages.Error as error:
```

Comment:
It would be good to have a comment here to explain what is happening. i.e. Some domains may be rejected during order creation and Certbot can retry the operation without the rejected domains.